### DISCUSSION TITLE
Anyone used zuluscsi on a Sun 3/60?

### DISCUSSION THREAD

**Discussion by twistdroach** [ZuluSCSI/ZuluSCSI-firmware#172] [General]
This is an ancient box - wonder if anyone has any hints on configuration?

I have tried many combinations of settings, but the best I can get is the following output from the boot prom (meaning - it's immediate, not loading boot loader or OS):
`sd: sense key = 3 error = 11, block 0`

Any advice on troubleshooting?  The log (even with debug mode turned on), doesn't seem to give any useful info.  The image I am using is a dd img of my working SunOS v4.1.1 os from an (also ancient) Micropolis 1684-7.  SunOS dmesg says this about the device:
`sd3: <Micropolis 1684-7 cel 1773 alt 2 hd 7 sec 53>`

I am working on updating the boot prom to a newer version which may have some better diagnostics...

**Comment by aperezbios** [comment]
@twistdroach what would be helpful is a debug log from the ZuluSCSI itself. Which model/form factor of ZuluSCSI are you attempting to use with the 3/60?

**Reply by twistdroach** [reply]
Here's the log output:
```
[10ms] Platform: ZuluSCSI RP2040
[10ms] FW Version: 23.03.30-release Mar 30 2023 16:53:18
[11ms] DIP switch settings: debug log 0, termination 1
[11ms] SCSI termination is enabled
[12ms] Flash chip size: 4096 kB
[14ms] SCSI target/disk mode selected by DIP switch, acting as a SCSI disk
[71ms] SD card detected, FAT32 volume size: 7572 MB
[72ms] SD MID: 0x03, OID: 0x53 0x44
[72ms] SD Name: SS08G
[73ms] SD Date: 7/2020
[73ms] SD Serial: 0xC3B9C671
[74ms] Reading configuration from zuluscsi.ini
[74ms] Active configuration:
[75ms] -- SelectionDelay: 0
[76ms] -- EnableSCSI2 is on
[78ms] -- EnableParity is on
[173ms] Finding HDD images in directory /:
[174ms] -- Opening /HD1.img for id:1 lun:0
[406ms] ---- WARNING: file /HD1.img is not contiguous. This will increase read latency.
[407ms] ---- Read prefetch enabled: 8192 bytes
[407ms] -- Opening /HD0.img for id:0 lun:0
[440ms] ---- WARNING: file /HD0.img is not contiguous. This will increase read latency.
[441ms] ---- Read prefetch enabled: 8192 bytes
[447ms] -- Platform supports ROM drive up to 3740 kB
[447ms] ---- ROM drive image not detected
[448ms] SCSI ID:0 BlockSize:512 Type:0 Quirks:0 ImageSize:329077kB
[448ms] SCSI ID:1 BlockSize:512 Type:0 Quirks:0 ImageSize:2097152kB
[549ms] Initialization complete!
[3914ms] DBG BUS RESET
[8909ms] DBG ---- SELECTION: 0
[8910ms] DBG ---- COMMAND: TestUnitReady
[8910ms] DBG ------ OUT: 0x00 0x00 0x00 0x00 0x00 0x00 
[8911ms] DBG ---- STATUS: 0 GOOD
```

And the ini looks like the following:
```

[SCSI]
# Settings that apply to all SCSI ids

# Select a system preset to apply default settings
# Known systems: "Mac", "MacPlus"
#System="Mac"

Debug = 1   # Same effect as DIPSW2, enables verbose log messages
SelectionDelay = 0   # Millisecond delay after selection, 255 = automatic, 0 = no delay
#SelectionDelay = 255   # Millisecond delay after selection, 255 = automatic, 0 = no delay
#PhyMode = 1   # 0: Best available  1: PIO  2: DMA_TIMER  3: GREENPAK_PIO   4: GREENPAK_DMA
#Dir = "/"   # Optionally look for image files in subdirectory
#Dir2 = "/images"  # Multiple directories can be specified Dir1...Dir9
#DisableStatusLED 1 # 0: Use status LED, 1: Disable status LED

# Settings that can be needed for compatibility with some hosts
#Quirks = 0   # 0: Standard, 1: Apple, 2: OMTI, 4: Xebec, 8: VMS
#EnableUnitAttention = 0 # Post UNIT_ATTENTION status on power-on or SD card hotplug
#EnableSCSI2 = 0 # Enable faster speeds of SCSI2
#EnableSelLatch = 1 # For Philips P2000C and other devices that release SEL signal before BSY
#EnableParity = 0 # Enable parity checks on platforms that support it (RP2040)
#MapLunsToIDs = 0 # For Philips P2000C simulate multiple LUNs
#MaxSyncSpeed = 10 # Set to 5 or 10 to enable synchronous SCSI mode, 0 to disable
MaxSyncSpeed = 0 # Set to 5 or 10 to enable synchronous SCSI mode, 0 to disable

# ROM settings
#DisableROMDrive = 1 # Disable the ROM drive if it has been loaded to flash
#ROMDriveSCSIID = 7 # Override ROM drive's SCSI ID

# Settings that can be specified either per-device or for all devices.
#Vendor = "QUANTUM"
#Product = "FIREBALL1"
#Version = "1.0"
#Serial = "0123456789ABCDEF"
#Type = 0     # 0: Fixed, 1: Removable, 2: Optical, 3: Floppy, 4: Mag-optical, 5: Tape
#TypeModifier = 0  # Affects only INQUIRY response
#SectorsPerTrack = 63
#HeadsPerCylinder = 255
#RightAlignStrings = 0 # Right-align SCSI vendor / product strings, defaults on if Quirks = 1
#PrefetchBytes = 8192 # Maximum number of bytes to prefetch after a read request, 0 to disable
#ReinsertCDOnInquiry = 1 # Reinsert any ejected CD-ROM image on Inquiry command

[SCSI0]
Vendor = "Micropolis"
Product = "1684-7"
SectorsPerTrack = 54
HeadsPerCylinder = 7

# Settings can be overridden for individual devices.
#[SCSI2]
#Product = "Disk with ID2"

#[SCSI5]
#Product = "CD-ROM Drive"
#Type = 2

# If IMG0..IMG9 are specified, they are cycled after each CD eject command.
#IMG0 = FirstCD.iso
#IMG1 = SecondCD.iso

# Raw sector range from SD card can be passed through
# Format is RAW:first_sector:last_sector where sector numbers can be decimal or hex.
# If end sector is beyond end of SD card, it will be adjusted automatically.
# [SCSI4]
# IMG0 = RAW:0x00000000:0xFFFFFFFF # Whole SD card
```

**Reply by twistdroach** [reply]
Oh- and it's a ZuluSCSI Compact RP2040 rev 2022a

**Reply by twistdroach** [reply]
Interesting data point - remembered that I had a Sun IPC box with a ZuluSCSI v1.1 in it - I copied the HD image onto the the ZS1.1 & the Sun 3/60 booted right up!  So must be some difference in the way the ZSv1.1 works vs the ZS Compact RP2040.  No ini file needed on the ZSv1.1.

**Reply by PetteriAimonen** [reply]
Interesting indeed, in #128 there has also been observed difference between v1.1 and RP2040. I haven't been able to figure out what it actually is.

The sense key 3-11 is [medium error, read error](https://en.wikipedia.org/wiki/Key_Code_Qualifier) but there is no read operation in the log.

If you are interested in testing, you could connect an USB cable to the RP2040 board and use a serial terminal program on PC to monitor the log output in real time. There is some chance that it is stuck mid-request, and the log file is only written after request ends or watchdog timer activates after 30 seconds.

**Comment by twistdroach** [comment]
Well - I think it was power related.  I was powering the board through termination power, but when I attached the USB, the system was able to boot right up.  Must have had just enough power to seem like it was working, but then reset or was momentarily unoperational once it was read from.  Anyway, it works fine once attached to a usb power source.  Thanks for the quick responses on this stuff, sorry for the run around.

**Reply by PetteriAimonen** [reply]
Nice to hear that it works now!

It might be worth it to implement some kind of voltage monitoring using the ADC in the future. The RP2040 processor will stay operational down to 1.8V supply, but as soon as the 3.0V supply falls a bit it starts affecting SD card and SCSI bus.
